pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -704,6 +725,38 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 	return changed
 }
 
+func delayPublishDomainConfig(ctx *zedmanagerContext, config types.AppInstanceConfig, status *types.AppInstanceStatus, dc *types.DomainConfig) {
+	status.State = types.START_DELAYED
+	uuidStr := status.Key()
+	passedSinceStart := time.Now().Sub(ctx.delayFromTime)
+	timer := time.NewTimer(config.StartDelay - passedSinceStart)
+	status.StartDelay = config.StartDelay
+	if ctx.delayFromTime.Add(config.StartDelay).After(ctx.delayWindowEnds) {
+		ctx.delayWindowEnds = ctx.delayFromTime.Add(config.StartDelay)
+		log.Warnf("@ohm: Added a new VM, extend the delay window (%s)!", ctx.delayWindowEnds.String())
+	}
+	ctx.appsInternalState[uuidStr] = types.AppInstanceInternalStatus{
+		UUID:       uuidStr,
+		DelayTimer: timer,
+	}
+	go func() {
+		log.Warnf("@ohm: Gonna wait now for %s (delay: %s, passed: %s, until: %s)",
+			(status.StartDelay - passedSinceStart).String(), status.StartDelay.String(), passedSinceStart.String(),
+			ctx.delayFromTime.Add(status.StartDelay).String())
+		<-timer.C
+		//nil the timer to prevent it from resetting on ConfigModify
+		log.Warnf("@ohm: Time has passed! Start the VM!")
+		nilTimerStatus := types.AppInstanceInternalStatus{UUID: uuidStr, DelayTimer: nil}
+		ctx.appsInternalState[uuidStr] = nilTimerStatus
```

Comment:
Shouldn't this delete the entry from the map?

Reply:
I should rethink this place, yep